runtime/ftplugin/help.lua

```diff
@@ -4,37 +4,45 @@ vim.treesitter.start()
 -- Add custom highlights for list in `:h highlight-groups`.
 local bufname = vim.fs.normalize(vim.api.nvim_buf_get_name(0))
 if vim.endswith(bufname, '/doc/syntax.txt') then
-  require('vim.vimhelp').highlight_groups({
+  require('_textutils').colorize_hl_groups({
     { start = [[\*group-name\*]], stop = '^======', match = '^(%w+)\t' },
     { start = [[\*highlight-groups\*]], stop = '^======', match = '^(%w+)\t' },
   })
 elseif vim.endswith(bufname, '/doc/treesitter.txt') then
-  require('vim.vimhelp').highlight_groups({
+  require('_textutils').colorize_hl_groups({
     {
       start = [[\*treesitter-highlight-groups\*]],
       stop = [[\*treesitter-highlight-spell\*]],
       match = '^@[%w%p]+',
     },
   })
 elseif vim.endswith(bufname, '/doc/diagnostic.txt') then
-  require('vim.vimhelp').highlight_groups({
+  require('_textutils').colorize_hl_groups({
     { start = [[\*diagnostic-highlights\*]], stop = '^======', match = '^(%w+)' },
   })
 elseif vim.endswith(bufname, '/doc/lsp.txt') then
-  require('vim.vimhelp').highlight_groups({
+  require('_textutils').colorize_hl_groups({
     { start = [[\*lsp-highlight\*]], stop = '^------', match = '^(%w+)' },
     { start = [[\*lsp-semantic-highlight\*]], stop = '^======', match = '^@[%w%p]+' },
   })
 end
 
 vim.keymap.set('n', 'gO', function()
-  require('vim.vimhelp').show_toc()
-end, { buffer = 0, silent = true })
+  require('_textutils').show_toc()
+end, { buffer = 0, silent = true, desc = 'Show table of contents for current buffer' })
+
+vim.keymap.set('n', ']]', function()
+  require('_textutils').jump({ count = 1 })
+end, { buffer = 0, silent = false, desc = 'Jump to next section' })
+vim.keymap.set('n', '[[', function()
+  require('_textutils').jump({ count = -1 })
+end, { buffer = 0, silent = false, desc = 'Jump to previous section' })
 
 -- Add "runnables" for Lua/Vimscript code examples.
 ---@type table<integer, { lang: string, code: string }>
 local code_blocks = {}
-local tree = vim.treesitter.get_parser():parse()[1]
+local bufnr = vim.api.nvim_get_current_buf()
+local parser = assert(vim.treesitter.get_parser(bufnr, 'vimdoc', { error = false }))
```

Comment:
```suggestion
local parser = assert(vim.treesitter.get_parser(nil, 'vimdoc', { error = false }))
```

The default is the current buffer.